src/ax_interface/socket_io.py

```diff
@@ -37,10 +52,21 @@ def __init__(self, mib_table, run_event, loop):
             try:
                 logger.info("Attempting AgentX socket bind...".format())
 
-                connection_routine = self.loop.create_unix_connection(
-                    protocol_factory=lambda: AgentX(self.mib_table, self.loop),
-                    path=constants.AGENTX_SOCKET_PATH,
-                    sock=self.ax_socket)
+                # Open the connection to the Agentx socket, we check the socket string to 
+                # either open a tcp socket or a Unix domain socket
+                if self.ax_socket_path.startswith('tcp'):
```

Comment:
>if self.ax_socket_path.startswith('tcp'): [](start = 16, length = 41)

Is domain socket good enough? [http://www.net-snmp.org/docs/man/snmpd.html](http://www.net-snmp.org/docs/man/snmpd.html) shows many other protocols. We can report 'not implemented' for others we don't expect.

Reply:
Can frr use domain socket?

---
In reply to: [234820946](https://github.com/Azure/sonic-snmpagent/pull/92#discussion_r234820946) [](ancestors = 234820946)

Reply:
frr can use a domain socket but that would involve mounting the domain socket from the host to all the containers. Instead of only modifying sonic-snmpagent we would also need to modify sonic-fpm-frr and sonic-buildimage to mount /var/agentx into the containers from the host.
This was actually my first implementation; but I found that going with the tcp port was way simpler and less intrusive in terms of modifications.

Reply:
implemented methods for udp, unix and tcp, also the default of udp port if only a number is passed

Reply:
I am ok with tcp socket. Still I think you don't need that much effort to use domain socket, and you can mount one folder in A docker to B docker.

---
In reply to: [234846409](https://github.com/Azure/sonic-snmpagent/pull/92#discussion_r234846409) [](ancestors = 234846409)

Reply:
I am not sure I understand your statement. but I mean "We can report 'not implemented' for others we don't expect.", such as ipx, ssh, etc.

---
In reply to: [235197136](https://github.com/Azure/sonic-snmpagent/pull/92#discussion_r235197136) [](ancestors = 235197136)